presto-iceberg/src/main/java/com/facebook/presto/iceberg/util/Statistics.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.iceberg.util;
+
+import com.facebook.presto.hive.metastore.HiveColumnStatistics;
+import com.facebook.presto.hive.metastore.PartitionStatistics;
+import com.facebook.presto.iceberg.IcebergColumnHandle;
+import com.facebook.presto.spi.ColumnHandle;
+import com.facebook.presto.spi.statistics.ColumnStatistics;
+import com.facebook.presto.spi.statistics.Estimate;
+import com.facebook.presto.spi.statistics.TableStatistics;
+
+import java.util.Map;
+
+public class Statistics
+{
+    private Statistics()
+    {
+    }
+
+    public static TableStatistics mergeHiveStats(TableStatistics icebergStats, PartitionStatistics hiveStats)
+    {
+        // We really only need to merge in NDVs and null fractions from the column statistics in hive's stats
+        //
+        // take iceberg data and column size, and row count statistics over hive's as they're more likely
+        // to be up to date since they are computed on the fly
```

Comment:
I didn't see a timestamp, but I think I agree that a method which chooses the most recent stats would be best. The reason that we override almost everything from Hive is that reading+merging from the parquet footers is most likely going to be up to date since they always (99% sure, but is there ever a case where they couldn't be?) represent the actual data, rather than relying on potentially out-of-date statistics from Hive.

The only ones which would potentially *not* be accurate are the ones which are not collected directly in the parquet footers: NDVs and the null fraction. Nulls counts are collected in the footers, but the row counts from iceberg are estimates, so it's possible nulls from Hive could end up being more accurate.

While I'm not entirely against making it configurable, I think that the cases where someone might want to override it is so small, I think it would not be worth doing so. Do you have a more general case in mind where someone might want to override the statistics from Hive over the parquet file-based ones?

Reply:
I think we should give users the option to tune how this feature gets applied. An enum flag like ICEBERG_HIVE_METASTORE_USE_TABLE_STATS with options {OFF, USE_NDV, USE_NULL_FRACTIONS, USE_ALL_STATS} would allow them more flexibility

Reply:
added